### PR TITLE
batches: Batch spec list improvements

### PR DESCRIPTION
**Pull request by eseliger**
This page was quickly created for debugging but we never got around putting some more effort into it. Timebox worked on this for a bit to make some improvements.

- Use the Duration component we already made public which has lots of improvements over this one, including live timers and fixed-width handling
- Improve state icons, a pending change should not be text-danger
- Added separate state for src-cli batch specs
- Tweaked wording of executed to created, because not all of them have been executed yet, kinda lazy fix, I know
- Fix execution time calculation, we used createdAt for it before which is incorrect, should be startedAt

## Test plan

Verified things still work and don't look worse than before.

## App preview:

- [Web](https://sg-web-es-batch-spec-list.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-plcioqbfzj.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/batches/BatchSpecNode.tsx [hunk 1]
```diff
@@ -114,7 +118,16 @@ export const BatchSpecNode: React.FunctionComponent<React.PropsWithChildren<Batc
     )
 }
 
-const StateIcon: React.FunctionComponent<React.PropsWithChildren<{ state: BatchSpecState }>> = ({ state }) => {
+const StateIcon: React.FunctionComponent<
+    React.PropsWithChildren<{ state: BatchSpecState; source: BatchSpecSource }>
+> = ({ state, source }) => {
+    if (source === BatchSpecSource.LOCAL) {
+        ;<Icon
```
Comment (BolajiOlajide) [inline client/web/src/enterprise/batches/BatchSpecNode.tsx:125]:
```suggestion
        <Icon
```

**Review by BolajiOlajide — APPROVED** [review]

File: client/web/src/enterprise/batches/BatchSpecNode.tsx (hunk 1 above)
Comment (eseliger) [reply client/web/src/enterprise/batches/BatchSpecNode.tsx:125]: oh dang! this is even missing a return :sweat:

Comment (eseliger) [inline client/web/src/enterprise/batches/BatchSpecNode.tsx:125]:
```suggestion
        return <Icon
```

**Review by adeola-ak — APPROVED** [review]

File: client/web/src/enterprise/batches/BatchSpecNode.tsx [hunk 2]
```diff
@@ -53,8 +54,11 @@ export const BatchSpecNode: React.FunctionComponent<React.PropsWithChildren<Batc
                 <Icon aria-hidden={true} svgPath={isExpanded ? mdiChevronDown : mdiChevronRight} />
             </Button>
             <div className="d-flex flex-column justify-content-center align-items-center px-2 pb-1">
-                <StateIcon state={node.state} />
-                <span className="text-muted">{upperFirst(node.state.toLowerCase())}</span>
+                <StateIcon source={node.source} state={node.state} />
+                <span className="text-muted">
+                    {node.source === BatchSpecSource.LOCAL && 'Uploaded'}
+                    {node.source !== BatchSpecSource.LOCAL && upperFirst(node.state.toLowerCase())}
```
Comment (courier-new) [inline client/web/src/enterprise/batches/BatchSpecNode.tsx:60]: Nice, love the "Uploaded" label, and admire your dedication to avoiding ternaries. 😂

File: client/web/src/enterprise/batches/BatchSpecNode.tsx [hunk 3]
```diff
@@ -146,32 +161,20 @@ const StateIcon: React.FunctionComponent<React.PropsWithChildren<{ state: BatchS
...
+                <Icon
+                    aria-hidden={true}
+                    className={classNames(styles.nodeStateIcon, 'text-muted mb-1')}
+                    svgPath={mdiPencil}
+                />
+            )
     }
 }
-
-const Duration: React.FunctionComponent<React.PropsWithChildren<{ start: Date; end: Date }>> = ({ start, end }) => {
```
Comment (courier-new) [inline client/web/src/enterprise/batches/BatchSpecNode.tsx:160]: Omg I didn't even realize this was still a separate component! Great refactor!

File: client/web/src/enterprise/batches/backend.ts [hunk 4]
```diff
@@ -86,8 +86,10 @@ const BATCH_SPEC_LIST_FIELDS_FRAGMENT = gql`
         __typename
         id
         state
+        startedAt
         finishedAt
         createdAt
```
Comment (courier-new) [inline client/web/src/enterprise/batches/backend.ts:91]: Are we using `createdAt` for anything now? Might be nice to drop it if not, save those 10 bytes or whatever haha.

**Review by courier-new — COMMENTED** [review]
Thanks for this, killer improvements. 🙂

File: client/web/src/enterprise/batches/BatchSpecNode.tsx (hunk 3 above)
Comment (eseliger) [reply client/web/src/enterprise/batches/BatchSpecNode.tsx:160]: me neither, saw it and was like.. didn't we make this reusable? 😬

File: client/web/src/enterprise/batches/backend.ts (hunk 4 above)
Comment (eseliger) [reply client/web/src/enterprise/batches/backend.ts:91]: oh yeah likely not

**Pull request merged**